src/token/WstEthL2Token.sol

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: Apache 2
+pragma solidity >=0.8.8 <0.9.0;
+
+import {OwnableUpgradeable} from
+    "openzeppelin-contracts-upgradeable/contracts/access/OwnableUpgradeable.sol";
+import {ERC20Upgradeable} from
+    "openzeppelin-contracts-upgradeable/contracts/token/ERC20/ERC20Upgradeable.sol";
+import {ERC20BurnableUpgradeable} from
+    "openzeppelin-contracts-upgradeable/contracts/token/ERC20/extensions/ERC20BurnableUpgradeable.sol";
+import {UUPSUpgradeable} from
+    "openzeppelin-contracts-upgradeable/contracts/proxy/utils/UUPSUpgradeable.sol";
+import {INttToken} from "@wormhole-foundation/native_token_transfer/interfaces/INttToken.sol";
+
+/// @title WstEthL2Token
+/// @notice A UUPS upgradeable token with access controlled minting and burning.
+contract WstEthL2Token is
+    INttToken,
+    UUPSUpgradeable,
+    ERC20Upgradeable,
+    ERC20BurnableUpgradeable,
+    OwnableUpgradeable
```

Comment:
Just as an aside we could use the `Ownable2StepUpgradeable` contract instead if we wanted, dealers choice

Reply:
The Lido team specifically requested to not use any 2-step upgrade processes.